app/gui/src/project-view/components/visualizations/TableVisualization.vue

```diff
@@ -205,13 +214,14 @@ function setRowLimit(newRowLimit: number) {
   }
 }
 
-watchEffect(() =>
+watchEffect(() => {
+  grid.value?.forceGridReRender()
```

Comment:
This immediately re-renders the grid once rendered first time. Please add a comment explaining why we need it.